plugin.rb

```diff
@@ -57,4 +57,12 @@
   Discourse::Application.routes.append do
     mount ::DiscourseSubscriptions::Engine, at: 's'
   end
+
+  add_to_serializer(:site, :show_campaign_banner) do
+    enabled = SiteSetting.discourse_subscriptions_enabled
+    campaign_enabled = SiteSetting.discourse_subscriptions_campaign_enabled
+    goal_met = Discourse.redis.get("subscriptions_goal_met_date")
+
+    enabled && campaign_enabled && (!goal_met || 7.days.ago <= Date.parse(goal_met))
```

Comment:
We should have a spec for this. Just to be extra super safe, we should also have a spec that ensures this doesn't blow up if the date stored in Redis is invalid (since this is in the site serializer).

Reply:
Added tests here -- also it was blowing up with an invalid date so I added some logic to fail gracefully in that case.